org.civicrm.reqangular/src/common/directives/time-amount-picker.directive.js

```diff
@@ -0,0 +1,140 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/modules/directives'
+], function (_, directives) {
+  'use strict';
+
+  directives.directive('timeAmountPicker', ['$templateCache', function ($templateCache) {
+    return {
+      scope: {
+        timeAmountPickerMinAmount: '<',
+        timeAmountPickerMaxAmount: '<',
+        timeAmountPickerInterval: '<',
+        timeAmountPickerValue: '='
+      },
+      restrict: 'E',
+      controllerAs: 'picker',
+      controller: ['$scope', timeAmountPickerController],
+      template: $templateCache.get('time-amount-picker.html')
+    };
+  }]);
+
+  timeAmountPickerController.$inject = ['$scope'];
+
+  function timeAmountPickerController ($scope) {
+    var interval, minAmount, maxAmount;
+    var vm = this;
+
+    vm.hoursOptions = [];
+    vm.minutesOptions = [];
+    vm.selectedHours = '';
+    vm.selectedMinutes = '';
+
+    vm.buildMinutesOptions = buildMinutesOptions;
+    vm.calculateSelectedValue = calculateSelectedValue;
+
+    (function init () {
+      parseInitialValue();
+      startWatching();
+    })();
+
+    /**
+     * Builds hours options
+     */
+    function buildHoursOptions () {
+      var hour = Math.floor(minAmount);
+
+      vm.hoursOptions = [];
+
+      while (hour <= maxAmount) {
+        vm.hoursOptions.push(hour);
+
+        hour++;
+      }
+    }
+
+    /**
+     * Builds minutes options
+     */
+    function buildMinutesOptions () {
+      var minute = 0;
+
+      vm.minutesOptions = [];
+
+      while (minute < 60) {
+        var skip =
+          vm.selectedHours !== '' &&
+          ((+vm.selectedHours === Math.floor(minAmount) && minute < minAmount % 1 * 60) ||
```

Comment:
I have no idea what's going on here tbh, can you please explain in the function description how this skips the value?

Reply:
Also wrapping `Math.floor(minAmount) && minute < minAmount % 1 * 60` into a variable would help.